### PR TITLE
Remove experimental flag from navigation and file blocks to use the Interactivity API (as private)

### PR DESCRIPTION
**Pull request by SantosGuillamot**
## What?
I opened a while ago a similar pull request to [remove the Interactivity API experimental flag](https://github.com/WordPress/gutenberg/pull/50941) completely. However, as the navigation and the file blocks have already been experimental for a while, and they are not adding new functionalities, I believe we can remove the experimental flag from them while keeping it for other blocks using the Interactivity API.

Keep in mind that it doesn't mean exposing the Interactivity API. It remains private only available for these blocks. This means that external blocks/plugins cannot use it, and we can keep experimenting and iterating.

## Why?
At this moment, users can only test the navigation and the file blocks using the Interactivity API if they activate the Experimental flag on the Gutenberg Experiments page.

## How?
These are the steps I took:

* I removed the conditional loading of the `script-loader.php` from the `load.php` file.
* I changed the `viewScript` field of the `block.json` of the Navigation and File blocks to read the interactivity.js file.
* I removed the old `view.js` files.
* I moved the directives directly to the `index.php` of the Image and File blocks.
* I added some logic to ensure the JS files are only loaded when they are necessary.
* I added some logic to load the directives conditionally.

Apart from that, I fixed a small bug where the `closeMenu` function was being triggered more than expected when pressing the ESC key when the menu was closed.

## Testing Instructions
* Test that the Navigation and the File blocks interactivity work as expected without the Experimental flag enabled.
* Test that the scripts are loaded conditionally:
    * In the navigation block: when the overlay is set to off, and the submenus options to "open on click" and "show arrow" are disabled, scripts and directives shouldn't be there.
    * In the file block: when the "Show inline embed" option is disabled, scripts and directives shouldn't be there.

### CODE REVIEW

**Review by luisherranz — CHANGES_REQUESTED** [review]
I've detected a couple of problems related to the files and directives:

- When there are no submenus, we should ignore `openSubmenusOnClick` and `showSubmenuIcon` and treat them as if they were `false`.
- When `overlayMenu` is `false`, `openSubmenusOnClick` is `false` but `showSubmenuIcon` is `true`, the directives are not being injected into the `nav`. I don't know why. Looking at the code I couldn't spot the problem.

<a href="https://www.loom.com/share/2bd82cfd63c3484e92895670be31c57e">
    
  </a>

https://www.loom.com/share/2bd82cfd63c3484e92895670be31c57e

Is there an easy way to know if there are any submenus?

**Comment by SantosGuillamot** [comment]
Thanks a lot for the in-depth review and the video 🙂 

> When overlayMenu is false, openSubmenusOnClick is false but showSubmenuIcon is true, the directives are not being injected into the nav. I don't know why. Looking at the code I couldn't spot the problem.

Oh, I think it is because of [this conditional](https://github.com/WordPress/gutenberg/blob/2e23de1a8e7095851eb05ec0269bac92e823cd8c/packages/block-library/src/navigation/index.php#L595-L601). I guess this could be solved adding the directives to that element or adding `data-wp-interactive` to the submenus as well. Any preference?

> Is there an easy way to know if there are any submenus?

First thing that comes to my mind is using the Tag Processor, checking if there is an element with the `has-child` class in the inner_blocks, but not sure if it is the best approach. I'm gonna think about it and research how this could be done.

**Comment by SantosGuillamot** [comment]
I just updated the code to solve the issues you mentioned. Let me know if that's okay 🙂

File: lib/experiments-page.php [hunk 1]
```diff
@@ -96,7 +96,7 @@ function gutenberg_initialize_experiments_settings() {
 		'gutenberg-experiments',
 		'gutenberg_experiments_section',
 		array(
-			'label' => __( 'Use the Interactivity API in File, Navigation and Image core blocks. It also enables the <a href="https://github.com/WordPress/gutenberg/issues/50029">Behaviors UI</a> in the Image block.', 'gutenberg' ),
+			'label' => __( 'Use the Interactivity API in Image core block. It also enables the <a href="https://github.com/WordPress/gutenberg/issues/50029">Behaviors UI</a> in the Image block.', 'gutenberg' ),
```
Comment (luisherranz) [inline lib/experiments-page.php]:
```suggestion
			'label' => __( 'Use the Interactivity API to enable the <a href="https://github.com/WordPress/gutenberg/issues/50029">Behaviors UI</a> in the Image block.', 'gutenberg' ),
```

**Review by luisherranz — APPROVED** [review]
LGMT now! Thanks, Mario 🙂

**Review by luisherranz — APPROVED** [review]
LGMT now! Thanks, Mario 🙂

**Pull request merged**